server/src/domain/facial-recognition/facial-recognition.services.ts

```diff
@@ -175,4 +198,12 @@ export class FacialRecognitionService {
 
     return true;
   }
+
+  ensureFaceThumbnailPath(ownerId: string, personId: string, extension: string): string {
+    let folderPath = this.storageCore.getFolderLocation(StorageFolder.THUMBNAILS, ownerId);
+    folderPath = join(folderPath, personId.substring(0, 2), personId.substring(2, 4));
+    this.storageRepository.mkdirSync(folderPath);
```

Comment:
This same logic exists in each service. Can we move it to the storage core instead? Also I think we can just pass `filename` in and substring that. In this case you can just pass in `${personId}.jpeg}`, since generating a filename is more dependent on the calling implementation than it is on the common code to generate directories.